### PR TITLE
Refactor for bus_man

### PR DESCRIPTION
**Pull request by Aram-Anderson**
#

Description of Changes Made: 
Cleaning up auth user methods and getting bus_man views tests to pass
Anticipated Code Effects: 
(Would this merge affect any other piece of our code?)
None(???)
Reviewer: 
@djtrujillo @snayrouz @sidewinder2020

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -52,12 +52,12 @@ def self.user_quantity_of_items_ordered
     group(:email).joins(orders: :order_items).sum(:quantity)
   end
 
-  def business_manager?(store_id)
-    !self.user_roles.where("role_id = 1 AND store_id = #{store_id}").empty?
+  def business_manager?
+    roles.exists?(role: "business_manager")
```
Comment (snayrouz) [inline app/models/user.rb:56]: cool cool cool

File: app/models/user.rb [hunk 2]
```diff
@@ -52,12 +52,12 @@ def self.user_quantity_of_items_ordered
...
   end
 
-  def business_admin?(store_id)
-    !self.user_roles.where("role_id = 2 AND store_id = #{store_id}").empty?
+  def business_admin?
+    roles.exists?(role: "business_admin")
```
Comment (sidewinder2020) [inline app/models/user.rb:60]: noice, moving away from booleans arrays ^^

Comment (snayrouz) [inline app/models/user.rb:60]: thanks for adding this. i didn't have this on my master :/

**Review by snayrouz — APPROVED** [review]
Approved!

**Review by djtrujillo — APPROVED** [review]
I know we need a manager dashboard but it makes sense to change these from dashboard/items to just /items

File: spec/factories/user.rb [hunk 3]
```diff
@@ -5,5 +5,6 @@
     email { Faker::Internet.email }
     password { Faker::Internet.password }
     address { Faker::Address.street_address }
+    phone {Faker::PhoneNumber.phone_number}
```
Comment (sidewinder2020) [inline spec/factories/user.rb:8]: excellent ~ this was a missing attr causing issues in a few instances

**Review by sidewinder2020 — COMMENTED** [review]
looks good, these changes simplify the code, and smooth out the wrinkles

**Pull request merged**